auto_nag/scripts/fuzzing_bisection_without_regressed_by.py

```diff
@@ -46,27 +92,89 @@ def get_bz_params(self, date):
             "f3": "longdesc",
             "o3": "casesubstring",
             "v3": "since this bug contains a bisection range, could you fill (if possible) the regressed_by field",
+            # TODO: Nag in the duplicate target instead, making sure it doesn't already have regressed_by.
+            "f4": "resolution",
+            "o4": "notequals",
+            "v4": "DUPLICATE",
             "emaillongdesc1": "1",
             "emailtype1": "exact",
             "email1": "bugmon@mozilla.com",
         }
 
-    def filter_bugs(self, bugs):
-        # Exclude bugs assigned to nobody.
-        bugs = {
-            bug["id"]: bug
-            for bug in bugs.values()
-            if not utils.is_no_assignee(bug["assigned_to_email"])
-        }
+    def find_regressor_or_needinfo_target(
+        self, bugs: dict[str, dict]
+    ) -> dict[str, dict]:
+        # Needinfo assignee when there is one.
+        for bug in bugs.values():
+            if not utils.is_no_assignee(bug["assigned_to_email"]):
+                bug["needinfo_target"] = {
+                    "mail": bug["assigned_to_email"],
+                    "nickname": bug["assigned_to_nickname"],
+                }
+
+        people = People.get_instance()
 
         # Exclude bugs that do not have a range found by BugMon.
         def comment_handler(bug, bug_id):
-            if not any(
-                "BugMon: Reduced build range" in comment["text"]
-                or "The bug appears to have been introduced in the following build range"
-                in comment["text"]
-                for comment in bug["comments"]
-            ):
+            range_found = False
+            # We start from the last comment just in case bugmon has updated the range.
+            for comment in bug["comments"][::-1]:
+                if (
+                    "BugMon: Reduced build range" in comment["text"]
+                    or "The bug appears to have been introduced in the following build range"
+                    in comment["text"]
+                ):
+                    range_found = True
+
+                    # Try to parse the regression range to find the regressor or at least somebody good to needinfo.
+                    pushlog_match = PUSHLOG_PAT.search(comment["text"])
+                    url = (
+                        pushlog_match.group(1).replace("pushloghtml", "json-pushes")
+                        + "&full=1&version=2"
+                    )
+                    r = requests.get(url)
+
+                    changesets = [
+                        changeset
+                        for push in r.json()["pushes"].values()
+                        for changeset in push["changesets"]
+                        if not all(
+                            is_ignorable_path(path) for path in changeset["files"]
+                        )
```

Comment:
I would use `any` instead of `not all`, in this way it does need to do the full iteration after finding the first not ignorable file. Also, it read more natural:
```suggestion
                        if any(
                            not is_ignorable_path(path) for path in changeset["files"]
                        )
```

Reply:
Done in f36225d.